contracts/libraries/TwabLib.sol

```diff
@@ -362,8 +362,15 @@ library TwabLib {
         _accountDetails.nextTwabIndex = uint24(
             RingBufferLib.nextIndex(_accountDetails.nextTwabIndex, MAX_CARDINALITY)
         );
-
-        _accountDetails.cardinality += 1;
+        
+        // Prevent the Account specific cardinality from exceeding the MAX_CARDINALITY.
+        // The ring buffer length is limited by MAX_CARDINALITY. IF the account.cardinality
+        // exceeds the max cardinality, new observations would be incorrectly set or the
+        // observation would be out of "bounds" of the ring buffer. Once reached the
+        // AccountDetails.cardinality will continue to be equal to max cardinality.
+        if(_accountDetails.cardinality < MAX_CARDINALITY) {
+            _accountDetails.cardinality += 1;
+        }
```

Comment:
I know this is simple code, but I'd love to have this tested.  It's mission-critical.

Can you:

1. extract lines 362 - 373 into a separate pure function that's called `pushObservation(AccountDetails, uint256 cardinality)` that returns a new Account details.
2. expose the function in the harness

Test that it correctly wraps and caps the cardinality

Reply:
if it's not clear: also use the pure function instead of the code at 362 - 373!

Reply:
Affirmative @asselstine  - great feedback for this one. Will test'ify this bit of code.